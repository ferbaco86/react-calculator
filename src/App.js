function App() {
  return (
    <div>
      <p>
        FIRST MILESTONE REACT CALCULATOR
      </p>
    </div>
  );
}

export default App;
