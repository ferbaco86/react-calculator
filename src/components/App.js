import ButtonPanel from './ButtonPanel';
import Display from './Display';
import calculate from '../logic/calculate';

function App() {
  return (
    <>
      <Display />
      <ButtonPanel />
    </>
  );
}

calculate(0, 0, '-');

export default App;
