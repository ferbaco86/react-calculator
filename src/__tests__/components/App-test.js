import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import {
  render, screen,
} from '@testing-library/react';
import App from '../../components/App';

it('renders correctly', () => {
  const tree = renderer.create(<BrowserRouter><App /></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Home Components', () => {
  test('Have a Title welcoming to the page', () => {
    render(<BrowserRouter><App /></BrowserRouter>);

    expect(screen.getByTestId('welcome')).toHaveTextContent('WELCOME TO OUR SITE');
  });
});
