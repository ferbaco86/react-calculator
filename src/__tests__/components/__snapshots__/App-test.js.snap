// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <nav
    className="sc-bdfBwQ eOkbzq"
  >
    <h1
      className="sc-gsTCUz joJsAy"
    >
      Math Magicians
    </h1>
    <div
      className="sc-dlfnbm XLyPB"
    >
      <a
        className="sc-hKgILt iUuVZI"
        href="/"
        onClick={[Function]}
      >
        Home 
      </a>
      <a
        className="sc-hKgILt iUuVZI"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator 
      </a>
      <a
        className="sc-hKgILt iUuVZI"
        href="/quotes"
        onClick={[Function]}
      >
        Quotes 
      </a>
    </div>
  </nav>,
  <section
    className="sc-eCssSg hzVQOI"
  >
    <h1
      className="sc-jSgupP gOOBPz"
      data-testid="welcome"
    >
      WELCOME TO OUR SITE
    </h1>
    <p
      className="sc-gKsewC dWFBOh"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
    <p
      className="sc-gKsewC dWFBOh"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
  </section>,
]
`;
